### PR TITLE
[TK-12]: Logic for login

### PR DESCRIPTION
**Pull request by bilalnaeem9**
## Jira: [TK-12](https://shop-go.atlassian.net/browse/TK-12?atlOrigin=eyJpIjoiNDIxMTdmODM3NmIyNGUzYmIxODdjMDE1NmVkYjcxMTIiLCJwIjoiaiJ9)

**Description**:
- password validation implemented as well as no emojis in password
- password length fixed to min of 6 along with 1 special character and 1 number
- textfield animation added(shake left/right and boarder red) for wrong password
- secure password implement

**How to test it?**
- test user "test99"
- test password "passWord12$
- password with no requirement triggers animation

### CODE REVIEW

**Review by Annas90 — DISMISSED** [review]

**Review by haroonmaqsoodch — COMMENTED** [review]
create a mock username and password to test the logic

**Review by ammadahmed22 — COMMENTED** [review]
after entering the test credential and touching the Login button, the keyboard doesn't dismiss.

**Review by ammadahmed22 — COMMENTED** [review]
where is the username validation (like password validation )

File: Vape-X/LoginViewController.swift [hunk 1]
```diff
@@ -26,4 +30,49 @@ class LoginViewController: UIViewController {
...
+            } else {
+                txtfieldUsername.layer.borderColor = UIColor.red.cgColor
+                txtfieldUsername.layer.borderWidth = 1.0
+                txtfieldPassword.layer.borderColor = UIColor.red.cgColor
+                txtfieldPassword.layer.borderWidth = 1.0
+                txtfieldUsername.shake()
+                txtfieldPassword.shake()
+            }
+        }
+    }
```
Comment (ammadahmed22) [inline Vape-X/LoginViewController.swift:57]: make a func so when the username and password are wrong or right the textfield is reset (empty).

**Review by ammadahmed22 — CHANGES_REQUESTED** [review]
where is the username validation (like password validation )

File: Vape-X/LoginViewController.swift [hunk 2]
```diff
@@ -26,4 +30,49 @@ class LoginViewController: UIViewController {
...
+}
+
+extension UITextField {
```
[the remark this answers is not in the document]
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolved issue.

[the remark this answers is not in the document]
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolved issue.

File: Vape-X/LoginViewController.swift [hunk 3]
```diff
@@ -26,4 +30,49 @@ class LoginViewController: UIViewController {
...
+                } else {
+                    txtfieldUsername.layer.borderColor = UIColor.red.cgColor
+                    txtfieldUsername.layer.borderWidth = 1.0
+                    txtfieldPassword.layer.borderColor = UIColor.red.cgColor
+                    txtfieldPassword.layer.borderWidth = 1.0
+                    txtfieldUsername.shake()
+                    txtfieldPassword.shake()
+                }
+            } else {
+                txtfieldUsername.layer.borderColor = UIColor.red.cgColor
```
[the remark this answers is not in the document]
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolved issue.

[the remark this answers is not in the document]
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolved issue.

File: Vape-X/LoginViewController.swift [hunk 4]
```diff
@@ -26,4 +30,49 @@ class LoginViewController: UIViewController {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
     }
+    
+    func unWrapTextField() {
+        if let userNameText = txtfieldUsername.text, let passwordText = txtfieldPassword.text {
+            if passwordText.validPassword() {
```
[the remark this answers is not in the document]
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolved issue.

File: Vape-X/LoginViewController.swift [hunk 5]
```diff
@@ -26,4 +30,49 @@ class LoginViewController: UIViewController {
...
+        let animation = CABasicAnimation(keyPath: "position")
+        animation.duration = 0.05
+        animation.repeatCount = 5
+        animation.autoreverses = true
+        animation.fromValue = CGPoint(x: self.center.x - 4.0, y: self.center.y)
+        animation.toValue = CGPoint(x: self.center.x + 4.0, y: self.center.y)
+        layer.add(animation, forKey: "position")
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolved issue.

File: Vape-X/LoginViewController.swift (hunk 1 above)
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift:47]: resolved issue.

File: Vape-X/ValideteViewController.swift [hunk 6]
```diff
@@ -0,0 +1,8 @@
+//
+//  ValideteViewController.swift
+//  Vape-X
+//
+//  Created by Bilal Naeem on 10/5/22.
+//
+
+import Foundation
```
Comment (ammadahmed22) [inline Vape-X/ValideteViewController.swift:8]: What is this file?

File: Vape-X/LoginViewController.swift [hunk 7]
```diff
@@ -26,4 +31,44 @@ class LoginViewController: UIViewController {
...
+    
+    func unWrapTextField() {
+        if let userNameText = txtfieldUsername.text, let passwordText = txtfieldPassword.text {
+            if userNameText.validUser() && passwordText.validPassword() {
+                ifCorrect()
+            } else {
+                ifWrong()
+            }
+        }
+    }
```
Comment (ammadahmed22) [inline Vape-X/LoginViewController.swift:47]: You can combine this func and the one below (checkMocUserAndPassword)

File: Vape-X/ValideteViewController.swift (hunk 6 above)
Comment (bilalnaeem9) [reply Vape-X/ValideteViewController.swift:8]: this file is for extentions.

File: Vape-X/LoginViewController.swift (hunk 7 above)
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolve issue.

[the remark this answers is not in the document]
Comment (bilalnaeem9) [reply Vape-X/LoginViewController.swift]: resolve issue.

File: Vape-X/ValideteViewController.swift (hunk 6 above)
Comment (ammadahmed22) [reply Vape-X/ValideteViewController.swift:8]: Your file for an extension is called LoginValidateExtention this is an empty file called ValideteViewController

**Pull request closed without merging**